core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -489,15 +489,29 @@ impl SchedulerController {
         bank: &Bank,
     ) -> (u64, u64) {
         let cost = CostModel::calculate_cost(transaction, &bank.feature_set).sum();
-        let fee = bank.fee_structure.calculate_fee(
-            transaction.message(),
-            5_000, // this just needs to be non-zero
-            fee_budget_limits,
-            bank.feature_set
-                .is_active(&include_loaded_accounts_data_size_in_fee_calculation::id()),
-            bank.feature_set
-                .is_active(&remove_rounding_in_fee_calculation::id()),
-        );
+        let fee = if bank.feature_set.is_active(&reward_full_priority_fee::id()) {
```

Comment:
lets just rename this to reward. and on else arm, throw in a saturating_div(2)

Reply:
refactored at 389bad36287a363cca6e1c3465593106cae0a1f0 to move the whole calculating reward into bank, share same distribute/burn code to be consistent.